### PR TITLE
slightly optimized packages.rs

### PR DESCRIPTION
**Pull request by WilliamAnimate**
change `u32` to `i16`, optimized the counting algorithm (for speed/conciseness) at the expense of higher memory usage

# tl;dr: you pushed a minor oversight lmao L + skill issue + bozo + cope + ratio + counter ratio + checkmate + etc this isn't twitter/x/whatevertheyrecallingitnow

`u` is basically a float number, it would be **EXTREMELY SLIM** that you would have half a package (eg 873.5). we get packages by counting the lines printed by something like `pacman -Q`, and it should be impossible to print half a line

changed integer limits from 32 bit to 16 bit. its unlikely anyone would have more than 2^15-1 packages installed. are there even that many packages in the AUR + repositories?

changed the counting algorithm from a 3-liner to an one liner:

```rust
num_packages.lines().count() as i16
```

probably faster since it isn't in a loop counting every individual line?

> EDIT: this program may use more memory (more apparent when you have a TON of packages) when counting because it loads everything into memory and then counts it instead of incrementing a value over every line found. just found that out.

### CODE REVIEW

**Comment by WilliamAnimate** [comment]
@XandrCopyrighted you **cannot** live with this critical flaw in your git commit history, i am laughing at you so hard rn

**Comment by XandrCopyrighted** [comment]
I'm at school m8

**Comment by WilliamAnimate** [comment]
i am at school too, what's your point?

image

**Comment by XandrCopyrighted** [comment]
bloody...

**Comment by WilliamAnimate** [comment]
lets go free git green
